### ISSUE TITLE
NeoHdf5Io not reading ChannelIndex

### ISSUE THREAD

**Issue by mschachter** [NeuralEnsemble/python-neo#255]
I've been trying to use the new ChannelIndex objects from the master branch, and write them to a file using NeoHdf5Io. NeoHdf5Io writes Block, Segment, and ChannelIndex objects to the hdf5 file, but does not seem to be able to read the ChannelIndex objects. Here's some code that reproduces the problem:

```

import numpy as np

import neo
import quantities as pq

blk = neo.Block(name='The Block')
seg = neo.Segment(name='The Segment')
blk.segments.append(seg)

sig = neo.AnalogSignal(np.random.randn(16, 10000),
                       sampling_rate=1e3*pq.Hz,
                       t_start=0*pq.second,
                       units=pq.volt,
                       dtype='float32')

chx = neo.ChannelIndex(name='Electrode Array', index=range(16))
chx.analogsignals.append(sig)
seg.analogsignals.append(sig)
blk.channel_indexes.append(chx)

blk.create_many_to_one_relationship(force=True, recursive=True)
blk.create_many_to_many_relationship()

# write a new file
f = neo.io.NeoHdf5IO('/tmp/neo_file.h5')
f.write_all_blocks([blk])
f.close()

# read the file
f = neo.io.NeoHdf5IO('/tmp/neo_file.h5')
blocks = f.read_all_blocks(lazy=False, cascade=True)
f.close()

blk = blocks[0]

assert len(blk.channel_indexes) > 0, "Block should have one ChannelIndex"
```

**Comment by apdavison** [comment]
The original `NeoHdf5IO` was not compatible with the master branch; I have just merged a new version (see #249), which is able to read data files produced by Neo 0.4 and earlier, but does not have write support.

The reasons for not implementing write support are given in #249 - in brief, IO classes for two other, more widely-supported, HDF5-based formats (NWB and NIX) are in development and it would be better to use one of these once they are finished.

**Comment by apdavison** [comment]
Note that `NIXIO` is now available in the master branch, and should be used in place of `NeoHdf5IO`.
